spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml
+import com.intel.analytics.bigdl.dataset.{DataSet, MiniBatch}
+import com.intel.analytics.bigdl.{Criterion, Module}
+import com.intel.analytics.bigdl.optim.Optimizer
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import org.apache.spark.ml.param.shared.{HasFeaturesCol, HasInputCols, HasLabelCol}
+import org.apache.spark.ml.param.{Param, ParamMap, Params}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.rdd.RDD
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * A wrapper of Optimizer to support fit() in ML Pipelines as an Estimator
+ * feature column name and label column name should be provided in training Dataframe
+ * Model to be trained, Feature size, label size, batch shape size must also be provided
+ * For details usage, please refer to example :
+ * [[com.intel.analytics.bigdl.example.MLPipeline.DLEstimatorLeNet]]
+ */
+class DLEstimator[@specialized(Float, Double) T: ClassTag]
+(override val uid: String = "DLEstimator")(implicit ev: TensorNumeric[T])
+  extends MLEstimator with HasFeaturesCol with HasLabelCol with DLDataParams[T] {
+
+  def setFeaturesCol(featuresColName: String): this.type = set(featuresCol, featuresColName)
+
+  def setLabelCol(labelColName : String) : this.type = set(labelCol, labelColName)
+
+  def validateParameters(): Unit = {
+    val params = this.extractParamMap()
+    require(isDefined(modelTrain),
+      "DLEstimator: model for optimization must not be null")
+    require(isDefined(batchShape),
+      "DLEstimator: batchShape for optimization must not be null")
+    require(isDefined(featuresCol),
+      "DLEstimator: features data must not be null")
+    require(isDefined(featureSize),
+      "DLEstimator: features size col must not be null")
+    require(isDefined(labelCol),
+      "DLEstimator: label data must not be null")
+    require(isDefined(labelSize),
+      "DLEstimator: label size must not be null")
+    require(isDefined(criterion),
+      "DLEstimator: criterion must not be null")
+  }
+
+  override def process(dataFrame: DataFrame): MlTransformer = {
+
```

Comment:
Just added the logic to check the schema (to check if input data frame are of array types for feature col and label col). Also checked the  API, estimator itself won't trigger transformschema, only pipleline does.